src/utils/traffic-mesh.js

```diff
@@ -40,19 +45,27 @@ const startForwardProxy = async ({ port, frameworkPort, functionsPort, publishDi
   }
 
   const { subprocess } = runProcess({ log, args })
+  const { forwarder, firstBundleReady } = forwardMessagesToLog({ log, subprocess })
 
   subprocess.on('close', process.exit)
   subprocess.on('SIGINT', process.exit)
   subprocess.on('SIGTERM', process.exit)
   ;['SIGINT', 'SIGTERM', 'SIGQUIT', 'SIGHUP', 'exit'].forEach((signal) => {
     process.on(signal, () => {
+      forwarder.close()
+
       const sig = signal === 'exit' ? 'SIGTERM' : signal
       subprocess.kill(sig, {
         forceKillAfterTimeout: PROXY_EXIT_TIMEOUT,
       })
     })
   })
 
+  // Wait until the first edge handlers bundle is ready
+  //
+  // In case of errors, this delays the startup process until the errors are fixed
+  await firstBundleReady
```

Comment:
Is this required? The server can still run even with bundle failure right?

Reply:
We won't run any edge handler until the first bundle is done. If the user then accesses the site there is a chance they'll see invalid / unexpected responses if an edge handler is supposed to run on a set of routes but actually doesn't.

This is more of a product decision, however, so therefore, your call. ✌️

Reply:
> We won't run any edge handler until the first bundle is done. If the user then accesses the site there is a chance they'll see invalid / unexpected responses if an edge handler is supposed to run on a set of routes but actually doesn't.

Will this also happen if the server was started successfully, but at a later stage bundling failed?
Do we fall back to previously bundled handlers or stop running all handlers?

Reply:
In that case we'll fall back to previously bundled handlers.

Reply:
Cool, so in that case I think we should keep the code as is due to the different semantics.